client/rest-high-level/src/test/java/org/elasticsearch/client/RequestTests.java

```diff
@@ -1319,6 +1332,14 @@ private static void setRandomTimeout(Consumer<String> setter, TimeValue defaultT
         }
     }
 
+    private static void setRandomFlatSettings(Consumer<Boolean> setter, Map<String, String> expectedParams) {
```

Comment:
maybe we can replace the existing `setRandomFlatSettings(GetIndexRequest request, Map<String, String> expectedParams)` method with this one?